jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -702,6 +702,13 @@ private InterpreterResult executeSql(String propertyKey, String sql,
               String results = getResults(resultSet,
                   !containsIgnoreCase(sqlToExecute, EXPLAIN_PREDICATE));
               interpreterResult.add(results);
+              try {
+                if (resultSet.isClosed()) {
+                  break;
```

Comment:
Yes, kind of thought about it, but since resultSet is closed; which means no other option can be performed on this resultSet, and hence break loop, enter into finally block and over there close both resultSet and statement.